lib/std/mem.zig

```diff
@@ -635,13 +635,70 @@ test "lessThan" {
 /// Compares two slices and returns whether they are equal.
 pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
     if (a.len != b.len) return false;
-    if (a.ptr == b.ptr) return true;
+    if (a.len == 0 or a.ptr == b.ptr) return true;
+
+    // No vectorization in stage2 x86_64 or x86
+    // https://github.com/ziglang/zig/issues/17748
+    if (builtin.zig_backend != .stage2_x86_64 and builtin.zig_backend != .stage2_x86) {
+        if (@typeInfo(T) == .Int and std.math.isPowerOfTwo(@bitSizeOf(T))) return eqlBytes(std.mem.sliceAsBytes(a), std.mem.sliceAsBytes(b));
```

Comment:
Could this be expanded to use `std.meta.hasUniqueRepresentation` (and `@sizeOf(Elem) > 0`)?

Reply:
I did not know that was a function, an interesting idea. It seems to do what we want. Do I understand correctly that it ensures `sliceAsBytes` will be a "perfect" slice? If so, then I agree this is a much better suited function for this purpose. I was exploring making things like packed structs work, but got a bit side-tracked.

Reply:
To add to that, `@sizeOf(Elem)` being 0 should actually be made a immediate `return true`, as there being no data means there is nothing to compare.

Reply:
So, I thought about it some more, and `hasUniqueRepresentation` is the function we want to use. Will finish testing a push shortly. Also to my comment about `u0` not being uniquely represented, I take that back. It doesn't have any padding nor unused bits so it technically fits the description.